### PR TITLE
GEOS-9865

### PR DESCRIPTION
**Pull request by al-tabb**
I have implemented a new feature which makes optional the 'WFSrequest.txt' file included in shapefile/zip output of WFS GetFeature request.  The parameter is toggled on/off on the WFSAdminPage.  This feature is useful for those who do not wish to include the request URL in the downloaded file for security reasons.

JIRA ticket GEOS-9865

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Comment by bradh** [comment]
- Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)

That doesn't appear to have worked out.  https://osgeo-org.atlassian.net/browse/GEOS-9865? shows a different title.

I think you'll need a CLA too.

**Comment by al-tabb** [comment]
I wasn't aware that the commit message had to match the JIRA title exactly.  I suppose I can re-commit with the proper title.  Can you please expand on what you mean by:
> You'll need to fix the formatting (e.g with mvn on the command line).

**Comment by bradh** [comment]
> I wasn't aware that the commit message had to match the JIRA title exactly. I suppose I can re-commit with the proper title. 

That part is from the checklist, which you marked as complete.

> Can you please expand on what you mean by:
> 
> > You'll need to fix the formatting (e.g with mvn on the command line).

Your build is failing QA checks , because you have mis-formatted files. Check what the QA results are.

**Comment by aaime** [comment]
@al-tabb before committing one should try and run a full maven build on the command line (not from the IDE), to make sure everything is still building propertly. That also has the side effect of running the code formatter and fix the source code in the process. 
Formatting is one of the QA checks, other tools are run if you specify "-Dqa" in your maven command line, see also:
https://docs.geoserver.org/latest/en/developer/qa-guide/index.html

**Comment by aaime** [comment]
Ah, you're free to update the commit title and force-push.

Also noticed
* "Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates" is checked, but I don't see documentation updates about it
* A test for the web page is present (good!) but one for the output format is not (bad)

File: src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java [hunk 1]
```diff
@@ -53,6 +53,13 @@ public void testChangesToValues() throws Exception {
         ft.submit("submit");
         wfs = getGeoServerApplication().getGeoServer().getService(WFSInfo.class);
         assertEquals("allowGlobalQueries = false", false, wfs.getAllowGlobalQueries());
+        // test includeWFSRequestDumpFile
+        tester.startPage(WFSAdminPage.class);
+        ft = tester.newFormTester("form");
+        ft.setValue("includeWFSRequestDumpFile", false);
+        ft.submit("submit");
+        wfs = getGeoServerApplication().getGeoServer().getService(WFSInfo.class);
+        assertEquals("includeWFSRequestDumpFile= false", false, wfs.getIncludeWFSRequestDumpFile());
```
Comment (aaime) [inline src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java:62]: Should have been 

```suggestion
        assertFalse("includeWFSRequestDumpFile= false", wfs.getIncludeWFSRequestDumpFile());
```

(a PMD check will likely fail this line, after you get formatting going)

File: src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java [hunk 2]
```diff
@@ -69,6 +76,16 @@ public void testApply() throws Exception {
...
+        //test Apply includeWFSRequestDumpFile
+        tester.startPage(WFSAdminPage.class);
+        ft = tester.newFormTester("form");
+        ft.setValue("includeWFSRequestDumpFile", true);
+        ft.submit("apply");
+        // did not switch
+        tester.assertRenderedPage(WFSAdminPage.class);
+        // value was updated
+        wfs = getGeoServerApplication().getGeoServer().getService(WFSInfo.class);
+        assertEquals("includeWFSRequestDUmpFile = true", true, wfs.getIncludeWFSRequestDumpFile());
```
Comment (aaime) [inline src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java:88]: Should have been:

```suggestion
        assertTrue("includeWFSRequestDumpFile= true", wfs.getIncludeWFSRequestDumpFile());
```

(a PMD check will likely fail this line, after you get formatting going)

File: doc/en/user/source/services/wfs/webadmin.rst [hunk 3]
```diff
@@ -114,6 +114,8 @@ is not selected, OGC WKT format will be used. If this checkbox is selected, ESRI
 
 Note: this requires an ``esri.properties`` file to be provided in the ``user_projections`` subdirectory of the GeoServer data directory. This may be obtained from the GeoTools EPSG extension.
 
+Selecting the :guilabel:`Include WFS request dump file` checkbox specifies if the file 'wfsrequest.txt' will be included in the Shapefile zip output. 'wfsreqeuest.txt' contains a dump of the full request URL used to get the Shapefile zip output.  If this checkbox is not selected, 'wfsrequest.txt' will not be included in the output. If this checkbox is selected, 'wfsrequest.txt' will be included in the output.
```
Comment (bradh) [inline doc/en/user/source/services/wfs/webadmin.rst:117]: typo in the second sentence. 'wfsreqeuest.txt' should be 'wfsrequest.txt'

File: doc/en/user/source/services/wfs/webadmin.rst [hunk 4]
```diff
@@ -143,4 +145,4 @@ For example on command line we can adjust adding parameter:
 	
 Or in Tomcat properties file (``{TOMCAT_HOME}/conf/catalina.properties``) adding the line:
 
-    org.geoserver.wfs.xml.entityExpansionLimit=50
\ No newline at end of file
+    org.geoserver.wfs.xml.entityExpansionLimit=50
```
Comment (bradh) [inline doc/en/user/source/services/wfs/webadmin.rst:148]: Guess this is EOL marker.

File: src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java [hunk 5]
```diff
@@ -85,7 +87,7 @@ public void testApply() throws Exception {
         tester.assertRenderedPage(WFSAdminPage.class);
         // value was updated
         wfs = getGeoServerApplication().getGeoServer().getService(WFSInfo.class);
-        assertEquals("includeWFSRequestDUmpFile = true", true, wfs.getIncludeWFSRequestDumpFile());
+        assertTrue("includeWFSRequestDUmpFile = true", wfs.getIncludeWFSRequestDumpFile());
```
Comment (bradh) [inline src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java:90]: Looks a bit strange, but I assume this is case insensitive...

File: src/wfs/src/test/java/org/geoserver/wfs/response/ShapeZipTest.java [hunk 6]
```diff
@@ -696,6 +733,7 @@ private void checkShapefileIntegrity(String[] typeNames, final InputStream in)
         while ((entry = zis.getNextEntry()) != null) {
             final String name = entry.getName();
             found.add(name);
+            System.out.println(name);
```
Comment (bradh) [inline src/wfs/src/test/java/org/geoserver/wfs/response/ShapeZipTest.java:736]: Possibly left-over debugging?

**Review by bradh — CHANGES_REQUESTED** [review]
Except for the typo in the docs, LGTM.

File: doc/en/user/source/services/wfs/webadmin.rst (hunk 3 above)
Comment (al-tabb) [reply doc/en/user/source/services/wfs/webadmin.rst:117]: Good catch, will correct this.

File: src/wfs/src/test/java/org/geoserver/wfs/response/ShapeZipTest.java (hunk 6 above)
Comment (al-tabb) [reply src/wfs/src/test/java/org/geoserver/wfs/response/ShapeZipTest.java:736]: Right you are.. will remove now.

File: src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java (hunk 5 above)
Comment (al-tabb) [reply src/web/wfs/src/test/java/org/geoserver/wfs/web/WFSAdminPageTest.java:90]: I'll correct this as well

**Comment by aaime** [comment]
Good progress! Don't forget to send the CLA to osgeo, and let us know when it's done.

**Comment by al-tabb** [comment]
> Good progress! Don't forget to send the CLA to osgeo, and let us know when it's done.

Will do.

**Comment by al-tabb** [comment]
I have just sent in my signed Individual Contributor License Agreement to OSGeo.

**Review by bradh — COMMENTED** [review]
Nothing more from me.

**Review by bradh — APPROVED** [review]

**Comment by aaime** [comment]
Nice, squash-merged into 34d0ee3fb47bb23044bbfdab2e8b68f3cd6df9e7
@a-tab not sure if you need to backport or not. Normally a new feature requires a "one month" cooldown before being backported to stable, but this one is pretty small, it's easy to argue it won't have negative side effect on the devel list, and get permission to backport sooner.

**Pull request merged**